schema.json

```diff
@@ -0,0 +1,252 @@
+{
+    "userenv": {
+	"type": "dict",
+	"required": true,
+	"excludes": "userenvs",
+	"schema": {
+	    "name": {
+		"type": "string",
+		"empty": false,
+		"required": true
+	    },
+	    "label": {
+		"type": "string",
+		"empty": false,
+		"required": true
+	    },
+	    "source": {
```

Comment:
I wonder if we shouldn't rename this so it is not confused with building from source.  Maybe something like "origin"?

Reply:
Agreed